cfn-resources/cloud-backup/cmd/resource/resource.go

```diff
@@ -0,0 +1,377 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	log "github.com/sirupsen/logrus"
+	matlasClient "go.mongodb.org/atlas/mongodbatlas"
+)
+
+var CreateRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ClusterName", "ProjectId"}
+var ReadRequiredFields = []string{"ProjectId", "ClusterName", "Id", "ApiKeys.PublicKey", "ApiKeys.PrivateKey"}
+var UpdateRequiredFields []string
+var DeleteRequiredFields = []string{"ProjectId", "ClusterName", "Id", "ApiKeys.PublicKey", "ApiKeys.PrivateKey"}
+var ListRequiredFields = []string{"ApiKeys.PublicKey", "ApiKeys.PrivateKey", "ClusterName", "ProjectId"}
```

Comment:
Can we consolidate this with what you have in [validator.go](cfn-resources/cloud-backup/cmd/validator_def/validator.go) (seems to be repeated logic)

Reply:
We have different fields required for each methods in different resources.
For Example,

Required Fields for Read Method in Cloud-Backup: {**"ProjectId", "ClusterName"**, "Id", "ApiKeys.PublicKey", "ApiKeys.PrivateKey"}
Required Fields for Read Method in Database-User: {"ApiKeys.PublicKey", "ApiKeys.PrivateKey", **"ProjectId", "DatabaseName", "Username"**}